framework/core/Utils.pm

```diff
@@ -257,6 +266,52 @@ sub check_vid {
     return {valid => 1, bid => $1, type => $2};
 }
 
+=pod
+  Utils::ensure_valid_bid(pid, bid)
+
+Ensure C<bid> represents a valid bug-id in project C<cid>, terminating with a
+helpful error message if not. A bug-id is valid for a project if the project
+exists and the bug-id both exists in the project and has not been deprecated.
+=cut
+sub ensure_valid_bid {
+    @_ == 2 or die $ARG_ERROR;
+    my $pid = shift;
+    my $bid = shift;
+    my $project = Project::create_project($pid);
+    my $project_dir = "$SCRIPT_DIR/projects/$pid";
+
+    if ( ! -e "${project_dir}" ) {
+        confess("Error: ${pid}-${bid} is a non-existent bug\n");
+    }
+
+    if ( ! -e "${project_dir}/trigger_tests/${bid}" ) {
+        confess("Error: ${pid}-${bid} is a non-existent bug\n");
+    }
+
+    my @bug_ids = $project->get_bug_ids;
+    if ( ! grep( /^$bid$/, @bug_ids) ) {
+        confess("Error: ${pid}-${bid} is a deprecated bug\n");
```

Comment:
Ultimately, we want to allow checkout of deprecated bugs. However, this will require finishing and merging #312. 

I would suggest we leave this as-is in this PR, merge these changes into #312, and then alter behavior in #312 to allow checkout of deprecated bugs.